superset-frontend/src/SqlLab/components/SaveDatasetModal.tsx

```diff
@@ -0,0 +1,183 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useState, FunctionComponent } from 'react';
+import { Radio, AutoComplete, Input } from 'src/common/components';
+import StyledModal from 'src/common/components/Modal';
+import Button from 'src/components/Button';
+import { styled } from '@superset-ui/core';
+import { RadioChangeEvent } from 'antd/lib/radio';
+
+interface SaveDatasetModalProps {
+  visible: boolean;
+  onOk: () => void;
+  onHide: () => void;
+  handleDatasetNameChange: (e: { target: { value: any } }) => void;
```

Comment:
is there an `HTMLFormInput` type or something similar that you can use here?

Reply:
or this shorthand for a event handler `React.FormEventHandler<HTMLInputElement>`